File: app/src/main/java/co/tpcreative/supersafe/model/ItemEntityModel.java
package co.tpcreative.supersafe.model;

import co.tpcreative.supersafe.common.entities.ItemEntity;

public class ItemEntityModel {
    public int id;
    public String originalName;
    public String thumbnailName;
    public boolean isSyncCloud;
    public boolean isSyncOwnServer;
    public String global_original_id;
    public String global_thumbnail_id;
    public String categories_local_id;
    public String categories_id;
    public int formatType;
    public String thumbnailPath;
    public String originalPath;
    public String mimeType;
    public String fileExtension;
    public int statusAction;
    public String items_id;
    public int degrees;
    public boolean thumbnailSync;
    public boolean originalSync;
    public int fileType;
    public String title;
    public String size;
    public boolean isDeleteLocal;
    public boolean isDeleteGlobal;

    public int statusProgress;
    public int deleteAction;
    public boolean isFakePin;
    public boolean isSaver;
    public boolean isExport;
    public boolean isWaitingForExporting;
    public int custom_items;
    public boolean isUpdate;

    /*Push data to local db*/
    public ItemEntityModel(ItemModel items){
        this.id = items.id;
        this.originalName = items.originalName;
        this.thumbnailName = items.thumbnailName;
        this.items_id = items.items_id;
        this.fileType = items.fileType;
        this.formatType = items.formatType;
        this.title = items.title;
        this.isSyncCloud = items.isSyncCloud;
        this.isSyncOwnServer = items.isSyncOwnServer;
        this.thumbnailSync = items.thumbnailSync;
        this.originalSync = items.originalSync;
        this.degrees = items.degrees;
        this.global_original_id = items.global_original_id;
        this.global_thumbnail_id = items.global_thumbnail_id;
        this.categories_id = items.categories_id;
        this.categories_local_id = items.categories_local_id;
        this.originalPath = items.originalPath;
        this.thumbnailPath = items.thumbnailPath;
        this.mimeType = items.mimeType;
        this.fileExtension = items.fileExtension;
        this.statusAction = items.statusAction;
        this.size = items.size;
        this.statusProgress = items.statusProgress;
        this.isDeleteLocal = items.isDeleteLocal ;
        this.isDeleteGlobal = items.isDeleteGlobal;
        this.deleteAction = items.deleteAction;
        this.isFakePin = items.isFakePin;
        this.isSaver = items.isSaver;
        this.isExport = items.isExport;
        this.isWaitingForExporting = items.isWaitingForExporting;
        this.custom_items = items.custom_items;
        this.isUpdate = items.isUpdate;
    }

    public ItemEntityModel(ItemEntity items){
        this.id = items.id;
        this.originalName = items.originalName;
        this.thumbnailName = items.thumbnailName;
        this.items_id = items.items_id;
        this.fileType = items.fileType;
        this.formatType = items.formatType;
        this.title = items.title;
        this.isSyncCloud = items.isSyncCloud;
        this.isSyncOwnServer = items.isSyncOwnServer;
        this.thumbnailSync = items.thumbnailSync;
        this.originalSync = items.originalSync;
        this.degrees = items.degrees;
        this.global_original_id = items.global_original_id;
        this.global_thumbnail_id = items.global_thumbnail_id;
        this.categories_id = items.categories_id;
        this.categories_local_id = items.categories_local_id;
        this.originalPath = items.originalPath;
        this.thumbnailPath = items.thumbnailPath;
        this.mimeType = items.mimeType;
        this.fileExtension = items.fileExtension;
        this.statusAction = items.statusAction;
        this.size = items.size;
        this.statusProgress = items.statusProgress;
        this.isDeleteLocal = items.isDeleteLocal ;
        this.isDeleteGlobal = items.isDeleteGlobal;
        this.deleteAction = items.deleteAction;
        this.isFakePin = items.isFakePin;
        this.isSaver = items.isSaver;
        this.isExport = items.isExport;
        this.isWaitingForExporting = items.isWaitingForExporting;
        this.custom_items = items.custom_items;
        this.isUpdate = items.isUpdate;
    }

    public ItemEntityModel(){
        this.thumbnailName = "";
        this.isSyncCloud = false;
        this.isSyncOwnServer = false;
        this.isUpdate = false;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/model/ItemEntityModel.kt
package co.tpcreative.supersafe.model
import co.tpcreative.supersafe.common.entities.ItemEntity

class ItemEntityModel {
    var id = 0
    var originalName: String? = null
    var thumbnailName: String?
    var isSyncCloud: Boolean
    var isSyncOwnServer: Boolean
    var global_original_id: String? = null
    var global_thumbnail_id: String? = null
    var categories_local_id: String? = null
    var categories_id: String? = null
    var formatType = 0
    var thumbnailPath: String? = null
    var originalPath: String? = null
    var mimeType: String? = null
    var fileExtension: String? = null
    var statusAction = 0
    var items_id: String? = null
    var degrees = 0
    var thumbnailSync = false
    var originalSync = false
    var fileType = 0
    var title: String? = null
    var size: String? = null
    var isDeleteLocal = false
    var isDeleteGlobal = false
    var statusProgress = 0
    var deleteAction = 0
    var isFakePin = false
    var isSaver = false
    var isExport = false
    var isWaitingForExporting = false
    var custom_items = 0
    var isUpdate: Boolean
    var isRequestChecking: Boolean

    /*Push data to local db*/
    constructor(items: ItemModel) {
        id = items.id
        originalName = items.originalName
        thumbnailName = items.thumbnailName
        items_id = items.items_id
        fileType = items.fileType
        formatType = items.formatType
        title = items.title
        isSyncCloud = items.isSyncCloud
        isSyncOwnServer = items.isSyncOwnServer
        thumbnailSync = items.thumbnailSync
        originalSync = items.originalSync
        degrees = items.degrees
        global_original_id = items.global_original_id
        global_thumbnail_id = items.global_thumbnail_id
        categories_id = items.categories_id
        categories_local_id = items.categories_local_id
        originalPath = items.getOriginal()
        thumbnailPath = items.getThumbnail()
        mimeType = items.mimeType
        fileExtension = items.fileExtension
        statusAction = items.statusAction
        size = items.size
        statusProgress = items.statusProgress
        isDeleteLocal = items.isDeleteLocal
        isDeleteGlobal = items.isDeleteGlobal
        deleteAction = items.deleteAction
        isFakePin = items.isFakePin
        isSaver = items.isSaver
        isExport = items.isExport
        isWaitingForExporting = items.isWaitingForExporting
        custom_items = items.custom_items
        isUpdate = items.isUpdate
        isRequestChecking = items.isRequestChecking
    }

    constructor(items: ItemEntity) {
        id = items.id
        originalName = items.originalName
        thumbnailName = items.thumbnailName
        items_id = items.items_id
        fileType = items.fileType
        formatType = items.formatType
        title = items.title
        isSyncCloud = items.isSyncCloud
        isSyncOwnServer = items.isSyncOwnServer
        thumbnailSync = items.thumbnailSync
        originalSync = items.originalSync
        degrees = items.degrees
        global_original_id = items.global_original_id
        global_thumbnail_id = items.global_thumbnail_id
        categories_id = items.categories_id
        categories_local_id = items.categories_local_id
        originalPath = items.originalPath
        thumbnailPath = items.thumbnailPath
        mimeType = items.mimeType
        fileExtension = items.fileExtension
        statusAction = items.statusAction
        size = items.size
        statusProgress = items.statusProgress
        isDeleteLocal = items.isDeleteLocal
        isDeleteGlobal = items.isDeleteGlobal
        deleteAction = items.deleteAction
        isFakePin = items.isFakePin
        isSaver = items.isSaver
        isExport = items.isExport
        isWaitingForExporting = items.isWaitingForExporting
        custom_items = items.custom_items
        isUpdate = items.isUpdate
        isRequestChecking = items.isRequestChecking
    }

    constructor() {
        thumbnailName = ""
        isSyncCloud = false
        isSyncOwnServer = false
        isUpdate = false
        isRequestChecking = false
    }
}